ui/src/app/lib/atlasmap-data-mapper/components/mapping/mapping.field.detail.component.ts

```diff
@@ -44,17 +44,18 @@ import { MappingModel, FieldMappingPair, MappedField } from '../../models/mappin
             </div>
         </ng-template>
 
-        <div class='fieldDetail' style="margin-bottom:5px;" *ngIf="mappedField"
-            [tooltip]="tolTemplate" placement="left">
+        <div class='fieldDetail' style="margin-bottom:5px;" *ngIf="mappedField">
             <label class="parentObjectName" *ngIf="displayParentObject()">
                 <i [attr.class]="getSourceIconCSSClass()"></i>
                 {{ getParentObjectName() }}
+                <span class="pficon pficon-info" [tooltip]="tolTemplate" placement="left"></span>
             </label>
             <div style="width:100%;">
                 <input type="text" id="input-{{isSource?'source':'target'}}-{{mappedField.field.getFieldLabel(false)}}"
```

Comment:
Regarding that `{{mappedField.field.getFieldLabel(false)}}` binding, function calls inside templates have a negative impact on performance down the line, since the Angular template engine will re-execute that function on each pass of the `ChangeDetection` cycle.

Reply:
The same applies to the entire block actually. `getSourceIconCSSClass()` and `getParentObjectName()` should be moved to the component controller and exposed by means of component fields, or even class getters.